### PR TITLE
GUI add/list/connect to network

### PR DESCRIPTION
**Pull request by nicboul**
A first step. The connection threads needs to report itself a callback
must be created to destroy it when disconnection is being request
via the GUI.

The logs are not reported either.
The status of the connection (IP, being reconnecting, etc) must be
reported to the GUI too.

### CODE REVIEW

File: src/agent.c [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	pthread_join(thread_start, NULL);
+	agent_fini();
+	free(netname);
+}
+
+void
+agent_thread_start(const char *name)
+{
+	struct sched_param	param;
+	int			policy = SCHED_RR;
```
Comment (si74) [inline src/agent.c:116]: Maybe just a comment here about why this policy is being used.

File: src/gui/accountsettings.cpp [hunk 2]
```diff
@@ -19,6 +19,8 @@
 
 #include "accountsettings.h"
 
+#include "../agent.h"
```
Comment (si74) [inline src/gui/accountsettings.cpp:22]: are relative filepaths like this okay?

File: src/gui/maindialog.cpp [hunk 3]
```diff
@@ -77,26 +74,38 @@ void MainDialog::NowRun()
...
 
-void MainDialog::slotToggleAutoConnect(int checked)
+void MainDialog::slotAddNetwork(void)
 {
+	QString	provcode;
+	QString	netname;
+	bool	ok;
+
+	provcode = QInputDialog::getText(this, NULL,
+		tr("Your provisioning key:"), QLineEdit::Normal, NULL, &ok);
```
Comment (si74) [inline src/gui/maindialog.cpp]: Tweak dialog to "Enter your provisioning key:"

File: src/gui/accountsettings.cpp (hunk 2 above)
Comment (nicboul) [reply src/gui/accountsettings.cpp:22]: yes, it shows the close relationship between that header and the GUI.

**Review by si74 — APPROVED** [review]
👍

**Pull request merged**